source/extensions/filters/network/dubbo_proxy/router/router_impl.h

```diff
@@ -24,16 +24,13 @@ class Router : public Tcp::ConnectionPool::UpstreamCallbacks,
                Logger::Loggable<Logger::Id::dubbo> {
 public:
   Router(Upstream::ClusterManager& cluster_manager) : cluster_manager_(cluster_manager) {}
-  ~Router() {}
+  ~Router() override = default;
 
   // DubboFilters::DecoderFilter
   void onDestroy() override;
   void setDecoderFilterCallbacks(DubboFilters::DecoderFilterCallbacks& callbacks) override;
-  Network::FilterStatus transportBegin() override;
-  Network::FilterStatus transportEnd() override;
-  Network::FilterStatus messageBegin(MessageType type, int64_t message_id,
-                                     SerializationType serialization_type) override;
-  Network::FilterStatus messageEnd(MessageMetadataSharedPtr metadata) override;
+
+  FilterStatus onMessageDecoded(MessageMetadataSharedPtr metadata, ContextSharedPtr ctx) override;
```

Comment:
`const ContextSharedPtr?`

Reply:
const cannot be used here because it changes the value of message_origin_buffer_ in the Context, as used in the onMessageDecoded function:
upstream_request_buffer_.move(ctx->message_origin_data(), ctx->message_size());